tumpasrc/__init__.py

```diff
@@ -279,6 +339,10 @@ def generate(self):
             subkeys_expiration=True,
         )
         self.update_ui.emit(newk)
+        success_dialog = QtWidgets.QMessageBox()
+        success_dialog.setWindowTitle("New key generated")
+        success_dialog.setText("Key generation successful!")
+        success_dialog.exec()
         self.hide()
```

Comment:
Can we hide the dialog first, and then show the success dialog?

Reply:
Done!